apps/crn-server/src/data-providers/users.data-provider.ts

```diff
@@ -347,6 +355,24 @@ export const parseGraphQLUserToDataObject = (
       : 'Guest';
   const teams = parseGraphQLUserTeamConnections(flatTeams || []);
 
+  const workingGroups = item.referencingWorkingGroupsContents
+    ? item.referencingWorkingGroupsContents.map((wg) => {
+        const wgRole = wg.flatData.leaders?.reduce((result, leader) => {
+          if (leader.user?.[0]?.id === item.id) {
+            return leader.role || 'Member';
+          }
+          return result;
+        }, 'Member');
+
+        return {
+          id: wg.id,
+          name: wg.flatData.title || '',
+          role: wgRole as 'Chair' | 'Project Manager' | 'Member',
+          active: !wg.flatData.complete,
+        };
+      })
+    : [];
```

Comment:
I was wondering if this could be rewritten as 

```
  const workingGroups = (item.referencingWorkingGroupsContents || []).map(
    (wg) => {
      const leader = wg.flatData.leaders?.find(
        (leader) => leader.user?.[0]?.id === item.id,
      );
      const wgRole = leader ? leader.role : 'Member';

      return {
        id: wg.id,
        name: wg.flatData.title || '',
        role: wgRole as 'Chair' | 'Project Manager' | 'Member',
        active: !wg.flatData.complete,
      };
    },
  );
```

so maybe it's more readable the way we find the working groups role.

But it's just a suggestion, you don't need to implement if you don't like it :)